core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -151,7 +297,48 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
     } else Array.empty[OpVectorColumnMetadata]
 
     val columns = categoricalColumns ++ textColumns
-    OpVectorMetadata(getOutputFeatureName, columns, Transmogrifier.inputFeaturesToHistory(inN, stageName))
+
+    val sensitive: Map[String, SensitiveFeatureInformation] =
+      nameIdentificationResults map { results: NameIdentificationResults =>
+        logDebug(s"treatAsName: [${results.isName.mkString(",")}]")
+        logDebug(s"predictedNameProb: [${results.predictedNameProbs.mkString(",")}]")
+        logDebug(s"bestFirstNameIndexes: [${results.bestFirstNameIndexes.mkString(",")}]")
+        logDebug(s"pctMale: [${results.pctMale.mkString(",")}]")
+        logDebug(s"pctFemale: [${results.pctFemale.mkString(",")}]")
+        logDebug(s"pctOther: [${results.pctOther.mkString(",")}]")
+
+        if (results.isName exists identity) {
+          logWarning(
+            """Hey! Some of your text columns look like they have names in them. Are you sure you want to build a
+              |model with that information in them? There could be potential bias as a result! Here's what I found:
+              |""".stripMargin
+          )
+          val problemColIndexes = results.isName.zipWithIndex.filter(_._1).map(_._2)
+          problemColIndexes foreach { index: Int =>
+            logWarning {
+              s"""Column Name: ${inN(index).name}
+              |Predicted Probability of Name: ${results.predictedNameProbs(index)}
+              |Percentage Likely Male Names: ${results.pctMale(index)}
+              |Percentage Likely Female Names: ${results.pctFemale(index)}
+              |Percentage Where No Gender Found: ${results.pctOther(index)}
+              |""".stripMargin
+            }
+          }
+        }
+
+        // Transform into the tuple from feature name to SensitiveFeatureInformation
+        inN.zipWithIndex map { case (feature: TransientFeature, index: Int) =>
```

Comment:
does it need to be created for every feature passed in? why not just a map from the feature names found to be sensitive to the information? very large metadata can have a cost